dae/dae/genomic_resources/reference_genome.py

```diff
@@ -187,13 +396,39 @@ def get_sequence(self, chrom, start, stop):
             + (start - 1) // self._index[chrom]["seqLineLength"]
         )
 
-        length = stop - start + 1
+        if stop is None:
+            length = self._index[chrom]["length"] - start + 1
+        else:
+            length = stop - start + 1
         line_feeds = 1 + length // self._index[chrom]["seqLineLength"]
 
         sequence = self._sequence.read(length + line_feeds).decode("ascii")
         sequence = sequence.replace("\n", "")[:length]
         return sequence.upper()
 
+    def pair_iter(self, region_size=1_000_000):
+        """
+        Iterate and yield nucleotide pairs in the genome.
+
+        The return value is a triple of the chromosome, the left and the right
+        nucleotide.
+        """
+        self._sequence.seek(0)
+        last_chrom = None
+        last_nuc = None
+        for chrom, start, end in self.split_into_regions(region_size):
+            sequence = self.get_sequence(chrom, start, end)
```

Comment:
This reads the whole region into memory. We could probably add a new method, `fetch` to the reference genome class that returns an iterator over the specified sequence. And make the `get_sequence` method use this new `fetch` method.